### ISSUE TITLE
block stuck half in palette

### ISSUE THREAD

**Issue by maryfries** [jmoenig/Snap--Build-Your-Own-Blocks#1738]
I had the SET block inside a block, and when I tried to drop it back in the palette, it got stuck like this.

Snap 4.0.10.2
Chrome 60.0.3080.0

![hanging-set-block-in-pallete](https://cloud.githubusercontent.com/assets/6549787/25455617/817ffdfe-2a9e-11e7-9ebc-f9e116412eba.gif)

It only showed in this palette—meaning that I didn't see it if I switched palettes, but I did see it again if I switched back to Control.

However, when I finally applied my changes to the block I had used SET in, the relic disappeared.

**Comment by maryfries** [comment]
I had this happen again today...

![image](https://user-images.githubusercontent.com/6549787/28833598-cf1b14e4-76ad-11e7-9911-211f228ecb5f.png)

**Comment by jmoenig** [comment]
I can't reproduce this, no matter how hard I try. Can you open a JS console next time this happens? Thanks!

**Comment by maryfries** [comment]
Yes. Also, snap just crashed completely on me with the same project. Any way to save you state information when that happens?

**Comment by jmoenig** [comment]
What do you mean with "Snap crashed completely"? Did the browser quit unexpectedly, or did you use Chrome and get an "Aw, Snap!" message? In this case you're probably low on memory. Measures: Use regular Blocks zoom on your retina display, and switch the display to "default" resolution. Did you use mobile Safari? Can you share the project? Are there a zillion unused blocks from multiple libraries in there? If so, try unloading unused ones.

**Comment by maryfries** [comment]
"Aw, Snap!"

Memory looks ok, and I've done nothing except reload that one window.

There are not a zillion unused blocks.

https://snap.berkeley.edu/snapsource/snap.html#present:Username=mfries&ProjectName=U3-Optional-OpArt

**Comment by jmoenig** [comment]
True, I can see that, thanks! (Well, there *are* a lot of unused blocks, but this should be okay). Thing is, Chrome won't ever tell you if it's low on memory, instead it'll just quit on you without any prior indication.

Beautiful project, btw!!

**Comment by maryfries** [comment]
Thanks! Project credit goes to Selim Tezel. :)

**Comment by jmoenig** [comment]
Selim creates incredibly beautiful projects!
